src/utils/popoverMixins.js

```diff
@@ -58,11 +59,30 @@ export default {
         trigger.setTriggerBy(this)
       }
     },
+    clearTimeout() {
+      clearTimeout(this.delayTimeout)
+      this.delayTimeout = null
+    },
     toggle (e) {
       let trigger = getFirstChild(this.$refs.trigger)
       if (e && this.trigger === 'contextmenu' && trigger === e.target) e.preventDefault()
-      if (!(this.show = !this.show)) {
+      console.log(e.type)
+      if (this.show) {
+        if (e.type === 'mouseleave') { // only delay closing for hover events
+          this.delayTimeout = setTimeout(() => {
+            this.show = false
+            this.clearTimeout()
+          }, 200)
+        } else if (e.type === 'mouseenter') {
+          this.clearTimeout()
+        } else {
+          this.clearTimeout()
+          this.show = false;
+        }
         return
+      } else {
```

Comment:
Add some comments for each of the if else statement blocks describing the scenarios.

Reply:
done

Reply:
- Start comments with a capital letter
- Preferred to have the comments at the first line of the conditional code block, rather than inline with the condition.

``` Javascript
if (x) {
  // Comment
  y = 5;
} else {
  // Comment 2
  y = 2;
}
```